xdsl/dialects/seq.py

```diff
@@ -106,6 +106,28 @@ class CompRegOp(IRDLOperation):
         "`:` type($input)"
     )
 
+    def __init__(
+        self,
+        input: SSAValue,
+        clk: SSAValue,
+        reset: tuple[SSAValue, SSAValue] | None = None,
+    ):
+        super().__init__(
+            operands=[
+                input,
+                clk,
+                reset[0] if reset is not None else None,
```

Comment:
you can use `reset.get(0)` here to avoid this spaghetti code

Reply:
It seems to be handling `reset = None` not `reset` being a dict without `0`. You could shorten to `(reset or {0: None})[0]` but frankly that doesn’t seem as readable.